src/miner.cpp

```diff
@@ -1214,9 +1215,15 @@ void ThreadStaker::operator()(std::vector<ThreadStaker::Args> args, CChainParams
             } catch (const std::runtime_error& e) {
                 LogPrintf("ThreadStaker: (%s) runtime error: %s\n", e.what(), operatorName);
 
-                // Could be failed TX in mempool, wipe mempool and allow loop to continue.
-                LOCK(cs_main);
-                mempool.clear();
+                if (nClearFlag.find(arg.operatorID) == nClearFlag.end()) {
```

Comment:
How about a map of CKeyId and int for consecutive failures? It's easy to track how many failure this way. 

Note: Also, reset to 0 on successful block.